src/feeds/FeedReader.ts

```diff
@@ -0,0 +1,144 @@
+import { MatrixClient } from "matrix-bot-sdk";
+import { BridgeConfigFeeds } from "../Config/Config";
+import { ConnectionManager } from "../ConnectionManager";
+import { FeedConnection } from "../Connections";
+import LogWrapper from "../LogWrapper";
+import { MessageQueue } from "../MessageQueue";
+
+import axios from "axios";
+import Parser from "rss-parser";
+
+const log = new LogWrapper("FeedReader");
+
+export class FeedError extends Error {
+    constructor(
+        public url: string,
+        public cause: Error,
+    ) {
+        super(`Error fetching feed ${url}: ${cause.message}`);
+    }
+}
+
+export interface FeedEntry {
+    feed: {
+        title: string,
+        url:   string,
+    },
+    title: string,
+    link:  string,
+}
+
+interface AccountData {
+    [url: string]: string[],
+}
+
+function stripHtml(input: string): string {
+    return input.replace(/<[^>]*?>/g, '');
+}
+
+export class FeedReader {
+    private observedFeedUrls: string[];
+    private seenEntries: Map<string, string[]> = new Map();
+    static readonly seenEntriesEventType = "uk.half-shot.matrix-hookshot.feed.reader.seenEntries";
+
+    constructor(
+        private config: BridgeConfigFeeds,
+        private connectionManager: ConnectionManager,
+        private queue: MessageQueue,
+        private matrixClient: MatrixClient,
+    ) {
+        const feedConnections = this.connectionManager.getAllConnectionsOfType(FeedConnection);
+        this.observedFeedUrls = feedConnections.map(c => c.feedUrl);
+        connectionManager.on('new-connection', c => {
+            if (c instanceof FeedConnection) {
+                log.info('New connection tracked:', c.feedUrl);
+                this.observedFeedUrls.push(c.feedUrl);
+            }
+        });
+
+        log.info('Loaded feed URLs:', this.observedFeedUrls);
+
+        void this.loadSeenEntries().then(() => {
+            return this.pollFeeds();
+        });
+    }
+
+    private async loadSeenEntries(): Promise<void> {
+        const accountData = await this.matrixClient.getAccountData<AccountData>(FeedReader.seenEntriesEventType).catch((err: any) => {
+            if (err.statusCode === 404) {
+                return {} as AccountData;
+            } else {
+                throw err;
```

Comment:
I'm not sure we're catching this. I suspect it might cause the application to crash as `loadSeenEntries` will throw and uncaught promises crash later versions of Node.

Reply:
Changed it to be non-throwing in 428ca024569010d0350663fcfa3a7b760ec17744